src/test/java/seedu/weme/model/meme/DescriptionTest.java

```diff
@@ -0,0 +1,37 @@
+package seedu.weme.model.meme;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.weme.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class DescriptionTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Description(null));
+    }
+
+    @Test
+    public void constructor_invalidDescription_throwsIllegalArgumentException() {
+        String invalidDescription = "";
+        assertThrows(IllegalArgumentException.class, () -> new Description(invalidDescription));
+    }
+
+    @Test
+    public void isValidDescription() {
+        // null weme
+        assertThrows(NullPointerException.class, () -> Description.isValidDescription(null));
+
+        // invalid addresses
```

Comment:
Change `addresses` to `descriptions`, similarly for the comment below. Maybe can use global search to ensure all `address` are replaced?